sale_exception_holidays_public/tests/test_sale_exception.py

```diff
@@ -0,0 +1,77 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from datetime import timedelta
+
+from odoo import Command, fields
+from odoo.exceptions import ValidationError
+from odoo.tests import TransactionCase
+
+
+class TestSaleException(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.holiday_model = cls.env["hr.holidays.public"]
+        cls.holiday_model_line = cls.env["hr.holidays.public.line"]
+
+        # Remove possibly existing public holidays that would interfer.
+        cls.holiday_model_line.search([]).unlink()
+        cls.holiday_model.search([]).unlink()
+
+        # Create holidays
+        today = fields.Date.today()
+        this_year = today.year
+        holiday_date = today + timedelta(days=10)
+        holiday_1 = cls.holiday_model.create(
+            {"year": this_year, "country_id": cls.env.ref("base.sl").id}
+        )
+        cls.holiday_model_line.create(
+            {"name": "holiday 5", "date": holiday_date, "year_id": holiday_1.id}
+        )
+
+        cls.holiday_date = holiday_date
+        cls.holiday_1 = holiday_1
+
+    def test_sale_order_exception(self):
```

Comment:
When working with tests and dates I generally like to use `freezegun` . It allows to freeze the date used when launching tests